### DISCUSSION TITLE
extract CSS with esbundle without the `bundle: true` option

### DISCUSSION THREAD

**Discussion by MakhBeth** [vanilla-extract-css/vanilla-extract#131] [Q&A] [answered]
Hi I'm building a react design-system where every single component is published on npm.

Given a structure like: 
```
src 
  Button.tsx
  Button.css.ts
```
my desired output is something like this:
```
dist
  Button.js
  Button.css
```

And inside the Button.js dist file the required CSS with a require statement: `require("./Button.css");`

But the esbundle plugin only exports the CSS file with the bundle options with true.
This is an issue because being a component library I don't want to include every dependency inside the bundle.

Any suggestion? Thank you :D

**Comment by mattcompiles** [comment, marked as answer]
Hey Bud 👋 , 

This is an `esbuild` limitation as plugins are only activated when `bundle` is true. You could bundle each file individually by walking the file system with a script but that would be pretty unideal.

**Reply by MakhBeth** [reply]
Thanks for the clarification 🤟
Thinking also on your other answer [here](https://github.com/seek-oss/vanilla-extract/discussions/102#discussioncomment-736211) I think that I will deliver only a JS version compiled with babel. I will experiment that. Thanks again 👍
